src/main.cpp

```diff
@@ -2851,6 +2857,7 @@ bool AcceptBlock(CBlock& block, CValidationState& state, CBlockIndex** ppindex,
     if (!fRequested) {  // If we didn't ask for it:
         if (pindex->nTx != 0) return true;  // This is a previously-processed block that was pruned
         if (!fHasMoreWork) return true;     // Don't process less-work chains
+        if (fTooFarAhead) return true;      // Block height is too high
```

Comment:
Really odd how we have this flag-based coding style here; I don't see why we can't just put the various tests in this if block.

That said, you might as well follow the style right now and fix it up later in another cleanup pull-req.